File: demo6_TableView/demo6_TableView/ViewController.m
//
//  ViewController.m
//  demo6_TableView
//
//  Created by ChenXin on 16/9/21.
//  Copyright © 2016年 ChenXin. All rights reserved.
//

#import "ViewController.h"
#import "DetailedViewController.h"

@interface ViewController ()

@end

@implementation ViewController

#pragma mark -- table data source --
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    if(section == 0) {
        return 5;
    }else {
        return 6;
    }
}

//an example for header view

- (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section {
    UIView * header = [[UIView alloc] init];
    header.backgroundColor = [UIColor colorWithRed:245.0/255.0 green:212.0/255.0 blue:217.0/255.0 alpha:1.0];
    return header;
}

- (nullable NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
    return @" ";
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    tableView.rowHeight = 60;
    NSString *cellID = [NSString stringWithFormat:@"cell_sec_%ld", (long)indexPath.section];
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellID];
    if( cell == nil ) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:cellID];
    }
    cell.textLabel.text = [NSString stringWithFormat:@"%ld", (long)indexPath.row];
    cell.detailTextLabel.text = [NSString stringWithFormat:@"Hi, this is section %ld, row %ld", (long)indexPath.section, (long)indexPath.row];
    
    if( indexPath.section == 0 ) {
        cell.imageView.image = [UIImage imageNamed:@"milkicon"];
    } else if( indexPath.section == 1) {
        cell.imageView.image = [UIImage imageNamed:@"cherryicon"];
    } else {
        cell.imageView.image = [UIImage imageNamed:@"umbrellaicon"];
    }
    return cell;
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 3;
}

#pragma mark -- table events --
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [self performSegueWithIdentifier:@"showDetail" sender:indexPath];
}



#pragma mark -- ViewController lifecycle --
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark -- view controller data passing --
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if( [segue.identifier isEqualToString:@"showDetail"] ) {
        DetailedViewController *dvc = segue.destinationViewController;
        dvc.indexPath = sender;
    }
}


@end

File: demo8_CoreData/demo8_CoreData/ViewController.swift
//
//  ViewController.swift
//  demo8_CoreData
//
//  Created by ChenXin on 2016/11/19.
//  Copyright © 2016年 ChenXin. All rights reserved.
//

import UIKit
import CoreData

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        print("i'm hdvsyu how are you")
        var context = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
        var row1:AnyObject = NSEntityDescription.insertNewObject(forEntityName: "Users", into: context)
        row1.setValue("female", forKey: "attribute1")
        row1.setValue("xcxnxn", forKey: "attribute2")
        var row2:AnyObject = NSEntityDescription.insertNewObject(forEntityName: "Users", into: context)
        row2.setValue("hdvsyu", forKey: "attribute2")
        do {
            try context.save()
        }catch {
            
        }
        
        print("run here")
        
   
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

